internal/resources/grafana/resource_report.go

```diff
@@ -317,102 +326,94 @@ func ReadReport(ctx context.Context, d *schema.ResourceData, meta interface{}) d
-	client, _, idStr := ClientFromExistingOrgResource(meta, d.Id())
+	client, _, idStr := OAPIClientFromExistingOrgResource(meta, d.Id())
 	id, err := strconv.ParseInt(idStr, 10, 64)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	report, err := schemaToReport(client, d)
+	report, err := schemaToReport(d)
 	if err != nil {
 		return diag.FromErr(err)
 	}
-	report.ID = id
 
-	if err := client.UpdateReport(report); err != nil {
+	if _, err := client.Reports.UpdateReport(id, &report); err != nil {
 		data, _ := json.Marshal(report)
 		return diag.Errorf("error updating the following report:\n%s\n%v", string(data), err)
 	}
 	return ReadReport(ctx, d, meta)
 }
 
 func DeleteReport(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	client, _, idStr := ClientFromExistingOrgResource(meta, d.Id())
+	client, _, idStr := OAPIClientFromExistingOrgResource(meta, d.Id())
 	id, err := strconv.ParseInt(idStr, 10, 64)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	err = client.DeleteReport(id)
+	_, err = client.Reports.DeleteReport(id)
 	diag, _ := common.CheckReadError("report", d, err)
 	return diag
 }
 
-func schemaToReport(client *gapi.Client, d *schema.ResourceData) (gapi.Report, error) {
-	id := int64(d.Get("dashboard_id").(int))
-	uid := d.Get("dashboard_uid").(string)
-	if uid == "" {
-		dashboards, err := client.DashboardsByIDs([]int64{id})
-		if err != nil {
-			return gapi.Report{}, fmt.Errorf("error getting dashboard %d: %v", id, err)
-		}
-		for _, dashboard := range dashboards {
-			if int64(dashboard.ID) == id {
-				uid = dashboard.UID
-				break
-			}
-		}
-		if uid == "" {
-			return gapi.Report{}, fmt.Errorf("dashboard %d not found", id)
-		}
-	}
-
+func schemaToReport(d *schema.ResourceData) (models.CreateOrUpdateConfigCmd, error) {
 	frequency := d.Get("schedule.0.frequency").(string)
-	report := gapi.Report{
-		Dashboards: []gapi.ReportDashboard{
-			{
-				Dashboard: gapi.ReportDashboardIdentifier{
-					UID: uid,
-				},
-			},
-		},
+	report := models.CreateOrUpdateConfigCmd{
 		Name:               d.Get("name").(string),
 		Recipients:         strings.Join(common.ListToStringSlice(d.Get("recipients").([]interface{})), ","),
 		ReplyTo:            d.Get("reply_to").(string),
 		Message:            d.Get("message").(string),
 		EnableDashboardURL: d.Get("include_dashboard_link").(bool),
 		EnableCSV:          d.Get("include_table_csv").(bool),
-		Options: gapi.ReportOptions{
+		Options: &models.ReportOptionsDTO{
 			Layout:      d.Get("layout").(string),
 			Orientation: d.Get("orientation").(string),
 		},
-		Schedule: gapi.ReportSchedule{
+		Schedule: &models.ScheduleDTO{
 			Frequency: frequency,
 			TimeZone:  "GMT",
 		},
-		Formats: []string{reportFormatPDF},
+		Formats: []models.Type{reportFormatPDF},
+	}
+
+	id := int64(d.Get("dashboard_id").(int))
+	uid := d.Get("dashboard_uid").(string)
+	if uid == "" {
+		// It triggers the old way to generate reports
+		report.DashboardID = id
```

Comment:
We aren't able to retrieve dashboards by id anymore. 

We need to set it in the deprecated `dashboard_id` to make it works.